overlord/snapstate/handlers.go

```diff
@@ -997,7 +1036,73 @@ func (m *SnapManager) undoRefreshAliasesV2(t *state.Task, _ *tomb.Tomb) error {
 
 	snapst.AutoAliasesDisabled = autoDisabled
 	snapst.Aliases = oldAliases
-	Set(st, snapName, snapst)
+	newSnapStates := make(map[string]*SnapState, 1+len(otherSnapDisabled))
+	newSnapStates[snapName] = snapst
+
+	// if we disabled other snap aliases try to undo that
+	var conflicting map[string]bool
+	var curOtherSnapStates map[string]*SnapState
+	if len(otherSnapDisabled) != 0 {
+		conflicting = make(map[string]bool, len(otherSnapDisabled))
+		curOtherSnapStates = make(map[string]*SnapState, len(otherSnapDisabled))
+		for otherSnap, otherDisabled := range otherSnapDisabled {
+			var otherSnapSt SnapState
+			err := Get(st, otherSnap, &otherSnapSt)
+			if err != nil {
+				return err
+			}
+			curOtherInfo, err := otherSnapSt.CurrentInfo()
+			if err != nil {
+				return err
+			}
+
+			curOtherSnapStates[otherSnap] = &otherSnapSt
+
+			autoDisabled := otherSnapSt.AutoAliasesDisabled
+			if otherDisabled.Disabled {
```

Comment:
because these are other snaps, not the main focus, we are not trying to remember the full state for them and put it back on undo, just remember what we did as a "delta" and do then a best-effort to do the reverse as much as it make sense vs the global state at undo